### PR TITLE
added disabled state to Select Component

### PR DESCRIPTION
**Pull request by robinvdv12**
### This PR:

proposed release: `patch`

**Checklist** 🛡
- [x] I have exported my addition from `src/index.ts` (check if not applicable)

**Adds** ✨
- a `disabled` state to the Select Component

### CODE REVIEW

File: src/components/Select/index.tsx [hunk 1]
```diff
@@ -16,6 +16,7 @@ type OptionBase = {
 };
 
 type StateType = {
+    disabled: boolean;
```
Comment (CarloPalinckx) [inline src/components/Select/index.tsx]: this shouldn't be in the state since it doesn't have to change internally.

**Comment by CarloPalinckx** [comment]
The design seems a bit off from the UX resource.

File: src/components/Select/story.tsx [hunk 2]
```diff
@@ -63,6 +63,7 @@ class Demo extends Component<{}, { value: string }> {
                 emptyText={text('emptyText', 'No results')}
                 onChange={this.handleChange}
                 options={object('options', options)}
+                disabled={boolean('disabled', true)}
```
Comment (nicokoelewijn) [inline src/components/Select/story.tsx]: Please move the disabled toggle above the options. Due to a bug in storybook the option textarea takes to much space, hiding the disabled toggle.

File: src/components/Select/style.tsx [hunk 3]
```diff
@@ -98,13 +111,17 @@ const StyledWindow = withProps<WindowProps, HTMLDivElement>(styled.div)`
...
-const StyledInput = styled.div`
+const StyledInput = withProps<InputProps>(styled.div)`
     box-sizing: border-box;
     width: 100%;
     padding: 0 0 0 12px;
     border: solid 1px ${({ theme }): string => theme.Select.input.borderColor};
-    background: ${({ theme }): string => theme.Select.input.background};
+    background: ${({ theme, disabled }): string =>
+        disabled
+            ? '#f5f6f7 repeating-linear-gradient( -45deg, #f0f1f2, #f0f1f2 10px, transparent 10px, transparent 20px )'
```
Comment (nicokoelewijn) [inline src/components/Select/style.tsx]: Please make this background theme-able

File: src/components/Select/story.tsx (hunk 2 above)
Comment (nicokoelewijn) [reply src/components/Select/story.tsx]: Also, disabled isn't the default value on any of the component's stories. Could you make this false?

**Pull request merged**